env0/resource_environment.go

```diff
@@ -441,8 +441,8 @@ func setEnvironmentSchema(ctx context.Context, d *schema.ResourceData, environme
 		d.Set("output", string(environment.LatestDeploymentLog.Output))
 	}
 
-	// Don't update this value for workflow environments - this value should always be 'false'.
-	if _, isWorkflow := d.GetOk("sub_environment_configuration"); !isWorkflow && environment.RequiresApproval != nil {
+	//nolint:staticcheck // https://github.com/hashicorp/terraform-plugin-sdk/issues/817
+	if _, exists := d.GetOkExists("approve_plan_automatically"); exists && environment.RequiresApproval != nil {
```

Comment:
only make changes to approve_plan_automatically if someone has set a value for it.
it's tri-boolean value.
undefined, true, and false.